modules/light-clients/06-solomachine/types/update.go

```diff
@@ -112,6 +123,49 @@ func (cs ClientState) UpdateState(ctx sdk.Context, cdc codec.BinaryCodec, client
 
 	cs.Sequence++
 	cs.ConsensusState = consensusState
+
+	clientStore.Set(host.ClientStateKey(), types.MustMarshalClientState(cdc, &cs))
+
+	// set default consensus height with header height
+	consensusHeight := smHeader.GetHeight()
+	if cs.ClientType() != exported.Localhost {
+		clientStore.Set(host.ConsensusStateKey(consensusHeight), types.MustMarshalConsensusState(cdc, consensusState))
+	} else {
+		consensusHeight = types.GetSelfHeight(ctx)
+	}
+
+	// TODO: Should be telemetry and events be emitted from 02-client?
+	// The clientID would need to be passed an additional arg here
```

Comment:
I think we need to make a decision on whether telemetry and events should remain to be emitted from `02-client` or from within the `ClientState` implementation. We may need to pass the client ID as an additional arg if we want to do this in the lightclient implementation, or otherwise we may need to return the consensus height. Any thoughts on this? cc. @colin-axner @seantking

Reply:
We should not require light client implementations to emit events. This decision informed the proposed design. We should allow light client to optionally emit their own events and telemetry. This is technically possible by constructing a `ClientMessage` which contains the clientID, but I'm not opposed to adding the clientID as an argument to the function since 02-client already has that information. It doesn't provide any usefulness outside of event emission/telemetry since the client store is prefixed by the clientID

Reply:
To expand further on why we should not require light client implementations to emit events:

Relayers make use of the events we have in 02-client. By passing this to light client implementations, we lose standardization and risk light client implementators making mistakes. Since the functionality called by 02-client is very verbose now, 02-client can emit appropriate events when doing updates or updates on misbehaviour

Reply:
Okay, great, that makes a lot of sense! Thanks, I'll remove the events and telemetry from here.